diff --git a/src/Twilio/Base/Page.cs b/src/Twilio/Base/Page.cs
index 6c31c57d..a28d26e6 100644
--- a/src/Twilio/Base/Page.cs
+++ b/src/Twilio/Base/Page.cs
@@ -61,7 +61,7 @@ namespace Twilio.Base
         private static string UrlFromUri(Domain domain, string region, string uri)
         {
             var b = new StringBuilder();
-            b.Append("https://").Append(domain);
+            b.Append("https://").Append(TwilioClient.GetDomain());
             
             if (!IsNullOrEmpty(region))
             {
@@ -170,4 +170,4 @@ namespace Twilio.Base
             );
         }
     }
-}
\ No newline at end of file
+}
diff --git a/src/Twilio/Http/Request.cs b/src/Twilio/Http/Request.cs
index 7f318aeb..354c379f 100644
--- a/src/Twilio/Http/Request.cs
+++ b/src/Twilio/Http/Request.cs
@@ -75,7 +75,7 @@ namespace Twilio.Http
             Method = method;
 
             var b = new StringBuilder();
-            b.Append("https://").Append(domain);
+            b.Append("https://").Append(domain == Domain.Api ? TwilioClient.GetDomain() : domain.ToString());
             if (!string.IsNullOrEmpty(region))
             {
                 b.Append(".").Append(region);
diff --git a/src/Twilio/Rest/Fax/V1/Fax/FaxMediaResource.cs b/src/Twilio/Rest/Fax/V1/Fax/FaxMediaResource.cs
index c25251ad..2e8480d9 100644
--- a/src/Twilio/Rest/Fax/V1/Fax/FaxMediaResource.cs
+++ b/src/Twilio/Rest/Fax/V1/Fax/FaxMediaResource.cs
@@ -27,7 +27,7 @@ namespace Twilio.Rest.Fax.V1.Fax
             return new Request(
                 HttpMethod.Get,
                 Rest.Domain.Fax,
-                "/v1/Faxes/" + options.PathFaxSid + "/Media/" + options.PathSid + "",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes/" + options.PathFaxSid + "/Media/" + options.PathSid + "",
                 client.Region,
                 queryParams: options.GetParams()
             );
@@ -97,7 +97,7 @@ namespace Twilio.Rest.Fax.V1.Fax
             return new Request(
                 HttpMethod.Get,
                 Rest.Domain.Fax,
-                "/v1/Faxes/" + options.PathFaxSid + "/Media",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes/" + options.PathFaxSid + "/Media",
                 client.Region,
                 queryParams: options.GetParams()
             );
@@ -236,7 +236,7 @@ namespace Twilio.Rest.Fax.V1.Fax
             return new Request(
                 HttpMethod.Delete,
                 Rest.Domain.Fax,
-                "/v1/Faxes/" + options.PathFaxSid + "/Media/" + options.PathSid + "",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes/" + options.PathFaxSid + "/Media/" + options.PathSid + "",
                 client.Region,
                 queryParams: options.GetParams()
             );
@@ -361,4 +361,4 @@ namespace Twilio.Rest.Fax.V1.Fax
         }
     }
 
-}
\ No newline at end of file
+}
diff --git a/src/Twilio/Rest/Fax/V1/FaxResource.cs b/src/Twilio/Rest/Fax/V1/FaxResource.cs
index fcf88205..a8e33e7d 100644
--- a/src/Twilio/Rest/Fax/V1/FaxResource.cs
+++ b/src/Twilio/Rest/Fax/V1/FaxResource.cs
@@ -88,7 +88,7 @@ namespace Twilio.Rest.Fax.V1
             return new Request(
                 HttpMethod.Get,
                 Rest.Domain.Fax,
-                "/v1/Faxes/" + options.PathSid + "",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes/" + options.PathSid + "",
                 client.Region,
                 queryParams: options.GetParams()
             );
@@ -155,7 +155,7 @@ namespace Twilio.Rest.Fax.V1
             return new Request(
                 HttpMethod.Get,
                 Rest.Domain.Fax,
-                "/v1/Faxes",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes",
                 client.Region,
                 queryParams: options.GetParams()
             );
@@ -306,7 +306,7 @@ namespace Twilio.Rest.Fax.V1
             return new Request(
                 HttpMethod.Post,
                 Rest.Domain.Fax,
-                "/v1/Faxes",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes",
                 client.Region,
                 postParams: options.GetParams()
             );
@@ -406,7 +406,7 @@ namespace Twilio.Rest.Fax.V1
             return new Request(
                 HttpMethod.Post,
                 Rest.Domain.Fax,
-                "/v1/Faxes/" + options.PathSid + "",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes/" + options.PathSid + "",
                 client.Region,
                 postParams: options.GetParams()
             );
@@ -478,7 +478,7 @@ namespace Twilio.Rest.Fax.V1
             return new Request(
                 HttpMethod.Delete,
                 Rest.Domain.Fax,
-                "/v1/Faxes/" + options.PathSid + "",
+                "/2010-04-01/Accounts/" + client.AccountSid + "/Faxes/" + options.PathSid + "",
                 client.Region,
                 queryParams: options.GetParams()
             );
@@ -657,4 +657,4 @@ namespace Twilio.Rest.Fax.V1
         }
     }
 
-}
\ No newline at end of file
+}
diff --git a/src/Twilio/Twilio.cs b/src/Twilio/Twilio.cs
index e6403598..7e5de56c 100644
--- a/src/Twilio/Twilio.cs
+++ b/src/Twilio/Twilio.cs
@@ -11,6 +11,7 @@ namespace Twilio
         private static string _username;
         private static string _password;
         private static string _accountSid;
+        private static string _domain;
         private static ITwilioRestClient _restClient;
 
         private TwilioClient() {}
@@ -96,6 +97,34 @@ namespace Twilio
         }
 
         /// <summary>
+        /// Set the client Domain
+        /// </summary>
+        /// <param name="domain">Domain</param>
+        public static void SetDomain(string domain)
+        {
+            if (domain == null)
+            {
+                throw new AuthenticationException("Domain can not be null");
+            }
+
+            if (domain != _domain)
+            {
+                Invalidate();
+            }
+
+            _domain = domain;
+        }
+
+        /// <summary>
+        /// Get the client Domain
+        /// </summary>
+        /// <returns>Domain</returns>
+        public static string GetDomain()
+        {
+            return _domain;
+        }
+
+        /// <summary>
         /// Get the rest client
         /// </summary>
         /// <returns>The rest client</returns>
@@ -106,10 +135,10 @@ namespace Twilio
                 return _restClient;
             }
 
-            if (_username == null || _password == null)
+            if (_username == null || _password == null || _domain == null)
             {
                 throw new AuthenticationException (
-                    "TwilioRestClient was used before AccountSid and AuthToken were set, please call TwilioClient.init()"
+                    "TwilioRestClient was used before AccountSid and AuthToken were set, please call TwilioClient.init() and TwilioClient.SetDomain"
                 );
             }
 
diff --git a/test/Twilio.Test/Twilio.Test.csproj b/test/Twilio.Test/Twilio.Test.csproj
index 92a29914..3f2b0842 100644
--- a/test/Twilio.Test/Twilio.Test.csproj
+++ b/test/Twilio.Test/Twilio.Test.csproj
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <RootNamespace>Twilio.Tests</RootNamespace>
-    <TargetFrameworks>netcoreapp2.0;net451;net35</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.0</TargetFrameworks>
     <RuntimeIdentifier Condition=" '$(TargetFramework)' == 'net451' ">win7-x86</RuntimeIdentifier>
     <RuntimeIdentifier Condition=" '$(TargetFramework)' == 'net35' ">win7-x86</RuntimeIdentifier>
   </PropertyGroup>
